### PR TITLE
[WIP] [Router/buildServerApp] In case of 1 GraphQL error, report that.

### PR DESCRIPTION
**Pull request by alloy**
Related to https://artsyproduct.atlassian.net/browse/PLATFORM-1343
Depends on https://github.com/artsy/metaphysics/pull/1460

This still needs tests and perhaps an extra sanity check, but seems to work well in my limited testing.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit eb9ef18b2cb687f8d1d059b0b57abeef18c46e67

https://deploy-preview-1855--artsy-reaction.netlify.com

File: src/Artsy/Router/buildServerApp.tsx [hunk 1]
```diff
@@ -134,8 +136,27 @@ export function buildServerApp(config: ServerRouterConfig): Promise<Resolve> {
...
-        console.error("[Artsy/Router/buildServerApp] Error:", error)
-        reject(error)
+        if (
+          error instanceof RRNLRequestError &&
+          error.res &&
+          error.res.errors &&
+          error.res.errors.length === 1 &&
+          error.res.errors[0].extensions &&
+          error.res.errors[0].extensions.httpStatusCodes &&
+          error.res.errors[0].extensions.httpStatusCodes.length === 1
```
Comment (alloy) [inline src/Artsy/Router/buildServerApp.tsx]: This can probably use our `get` utility.

Comment (pepopowitz) [reply src/Artsy/Router/buildServerApp.tsx]: Heads up that per a conversation in Slack, I'm going to pick this PR up - I'll write some tests, and convert this to use `get`.

**Comment by pepopowitz** [comment]
I attempted to add test coverage for this, but ran into a snag. 

When `relay-mock-network-layer` throws an error, it always throws a [`RelayMockNetworkLayerError`](https://github.com/1stdibs/relay-mock-network-layer/blob/863b2a57ff91b77f5bf8351dc110ac4612db46a4/RelayMockNetworkLayerError.js). This doesn't have the same shape as a `RRNLRequestError`. So to make a test pass, I think I'd have to update the code to check for *either* one of those types of errors, which doesn't really give confidence that the *real* path is working.

Another thing I could think would make this testable would be to submit a PR to the `relay-mock-network-layer` project to throw an actual `RRNLRequestError` (or maybe a type that you give it). I'm not sure if that jives with that project's vision, and would definitely want help from @alloy to do that.

If either of you @alloy or @mzikherman have some ideas on how to best handle this, I'd love to hear your thoughts (and/or pair)!

I'm also still planning on making the change to use `get` for that long if statement. That should be coming soon.

**Comment by pepopowitz** [comment]
Ok, I understand the differences now: `artist`/`artwork2` are hitting a restful endpoint at gravity, so they get actual 404s back. `order2` hits exchange, a _graphql endpoint_, so it gets a 200. I initially thought that this PR would also help us with the fact that we weren't seeing 404s when an offer is not found, but I no longer think that is the case.

**Comment by damassi** [comment]
@pepopowitz / @alloy - thoughts on moving this PR forward or closing?

**Comment by alloy** [comment]
This should not be closed, I’ve moved it to the platform board instead.

**Comment by alloy** [comment]
Sounds good 👌 Let’s discuss/do this!

**Comment by damassi** [comment]
@mzikherman - can this be closed?

**Comment by mzikherman** [comment]
Yes!

**Pull request closed without merging**